tests/tracing/conftest.py

```diff
@@ -0,0 +1,47 @@
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import SpanExporter, SimpleSpanProcessor
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import InMemorySpanExporter
+
+from mlserver import Settings, MLServer
+from pytest import fixture
+from typing import AsyncGenerator
+from mlserver.grpc.dataplane_pb2_grpc import GRPCInferenceServiceStub
+from grpc import aio
+
+
+@fixture
+def settings(settings: Settings) -> Settings:
+    settings.tracing_server = "something:port"
+    return settings
+
+
+@fixture
+def span_exporter(mocker) -> SpanExporter:
+    span_exporter = InMemorySpanExporter()
+    mocker.patch("mlserver.tracing._create_span_exporter", return_value=span_exporter)
+    yield span_exporter
+    span_exporter.clear()
+
+
+@fixture(autouse=True)
+def tracer_provider(mocker, span_exporter, settings) -> TracerProvider:
+    resource = Resource(attributes={"service.name.testing": "mlserver-testing"})
+    tracer_provider = TracerProvider(resource=resource)
+    span_processor = SimpleSpanProcessor(span_exporter)
+    tracer_provider.add_span_processor(span_processor)
+
+    mocker.patch("mlserver.rest.app.get_tracer_provider", return_value=tracer_provider)
+    mocker.patch(
+        "mlserver.grpc.server.get_tracer_provider", return_value=tracer_provider
+    )
+    yield tracer_provider
+    tracer_provider.shutdown()
```

Comment:
Out of curiosity, do we need to also shutdown the regular OTLP tracer?

Reply:
When the OTLP tracer is shut down, it sends a signal for every span processor to flush outstanding spans to their corresponding exporters, and for them in turn to stop handling any further incoming spans. I believe we don't need to shut it down manually, because the app is instrumented to run with tracing for its whole duration, if tracing is enabled. If at some point, we decide to implement a feature to disable tracing, via API for example, then I think it would make sense to do that basically to tell the tracer that we don't want to process and handle new spans anymore.

Reply:
Got it - as long as it cleans itself when MLServer shuts down we should be OK